src/reflect/scala/reflect/internal/util/ScalaClassLoader.scala

```diff
@@ -46,6 +47,35 @@ trait ScalaClassLoader extends JClassLoader {
   def create(path: String): AnyRef =
     tryToInitializeClass[AnyRef](path).map(_.newInstance()).orNull
 
+  /** Create an instance with ctor args, or invoke errorFn before throwing FatalError. */
+  def create[T <: AnyRef : ClassTag](path: String, errorFn: String => Unit)(args: AnyRef*): T = {
+    def fail(msg: String) = {
```

Comment:
I don't think we should use FatalError here. It's extremely poorly named, but it's meant to indicate a problem with class loading in the type checker/namer. Let's throw the original exception instead -- it'll be more helpful to diagnose weird failures (just spent a couple of hours the other day figuring out why a class def could not be found; instead, it was an exception encountered during the class's initializer.)